### PR TITLE
Define `_MDSPAN_NO_UNIQUE_ADDRESS` based on CXX version.

### LINKED ISSUE #161: Define `_MDSPAN_NO_UNIQUE_ADDRESS` based on CXX version.
Hi, we run into an issue with nvcc 1.6 and g++ 9.4 using c++ 17. The macro `_MDSPAN_NO_UNIQUE_ADDRESS` is defined differently by these two compilers. I opened a fix https://github.com/rapidsai/raft/pull/764, do you think it's a viable approach and would you like to have it in here as well?

### PR DESCRIPTION
**Pull request by trivialfis**
Close https://github.com/kokkos/mdspan/issues/161

For the changed test, please see: https://github.com/kokkos/mdspan/issues/161#issuecomment-1198879896 .

### CODE REVIEW

**Comment by mhoemmen** [comment]
@crtrott @dalg24 @nliber

**Review by dalg24 — APPROVED** [review]
Fine by me

File: compilation_tests/ctest_no_unique_address.cpp [hunk 1]
```diff
@@ -92,7 +92,7 @@ MDSPAN_STATIC_TEST(
   >) == sizeof(size_t)
 );
 
-#if defined(__GNUC__) && (__GNUC__>8)
+#if defined(__GNUC__) && (__GNUC__> 10)
```
Comment (crtrott) [inline compilation_tests/ctest_no_unique_address.cpp:95]: Maybe we should simply check _MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS here?

**Review by mhoemmen — APPROVED** [review]

**Review by nliber — APPROVED** [review]

File: include/experimental/__p0009_bits/config.hpp [hunk 2]
```diff
@@ -121,10 +121,8 @@ static_assert(_MDSPAN_CPLUSPLUS >= MDSPAN_CXX_STD_14, "mdspan requires C++14 or
 #  define _MDSPAN_PRESERVE_STANDARD_LAYOUT 1
 #endif
 
-// no unique address starts working in NVCC 11.6
-#if !defined(_MDSPAN_USE_ATTRIBUTE_NO_UNIQUE_ADDRESS) && \
-   (!defined(__NVCC__) || ((__CUDACC_VER_MAJOR__ > 11) && (__CUDACC_VER_MINOR__ > 5)))
-#  if (__has_cpp_attribute(no_unique_address) >= 201803L) && !defined(_MDSPAN_COMPILER_MSVC)
```
Comment (crtrott) [inline include/experimental/__p0009_bits/config.hpp:127]: this also remove the MSVC check ....

**Review by crtrott — COMMENTED** [review]
I think this was screwing up other compilers

**Pull request merged**